shared-bin/src/logger.rs

```diff
@@ -1,15 +1,20 @@
 use anyhow::anyhow;
 use anyhow::Result;
+use opentelemetry::sdk::propagation::TraceContextPropagator;
 use time::macros::format_description;
 use tracing_subscriber::fmt::time::UtcTime;
 use tracing_subscriber::EnvFilter;
+use tracing_subscriber::Layer;
+use tracing_subscriber::Registry;
 
 pub use tracing_subscriber::filter::LevelFilter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
 
 const RUST_LOG_ENV: &str = "RUST_LOG";
 
 #[allow(clippy::print_stdout)] // because the logger is only initialized at the end of this function but we want to print a warning
-pub fn init(level: LevelFilter, json_format: bool) -> Result<()> {
+pub fn init(level: LevelFilter, json_format: bool, name: &str) -> Result<()> {
```

Comment:
From the api perspective it is not clear what `name` is used for. If I understand it correctly, than it's the `jaeger service name`. Might be good to rename it.

Reply:
Good point

Reply:
`tracing_service_name` perhaps?